### ISSUE TITLE
libsel4vm: set VM_GUEST_ERROR_EXIT on vCPU index error?

### ISSUE THREAD

**Issue by axel-h** [seL4/seL4_projects_libs#47]
I wonder, in https://github.com/seL4/seL4_projects_libs/blob/9de06305edb99bbe73be1094ed1b9bc47679b854/libsel4vm/src/arch/arm/vm.c#L234-L241 shouldn't this also set
```
vm->run.exit_reason = VM_GUEST_ERROR_EXIT;
```
if vcpu_idx is invalid?

**Comment by Ivan-Velickovic** [comment]
Is there an instance where `exit_reason` is actually used? I can only see it being set.

**Comment by axel-h** [comment]
I though it's part of the API. So the examples should be improved to make use of this - or have a comment that they don't use this because it does not make an difference for the example.

**Comment by alisonfel** [comment]
> I wonder, in
> 
> https://github.com/seL4/seL4_projects_libs/blob/9de06305edb99bbe73be1094ed1b9bc47679b854/libsel4vm/src/arch/arm/vm.c#L234-L241
> shouldn't this also set
> 
> ```
> vm->run.exit_reason = VM_GUEST_ERROR_EXIT;
> ```
> 
> if `vcpu_idx` is invalid?

Yep, `vm->run.exit_reason = VM_GUEST_ERROR_EXIT;` should be set when the `vcpu_idx` is invalid. This is likely a small bug, I might've just missed this when refactoring the libraries a few years ago.

**Comment by axel-h** [comment]
I assume it should also be set here then: https://github.com/seL4/seL4_projects_libs/blob/9de06305edb99bbe73be1094ed1b9bc47679b854/libsel4vm/src/arch/arm/vm.c#L249-L255

**Comment by alisonfel** [comment]
Hmm, I don't follow? That case should be already handled just below:
https://github.com/seL4/seL4_projects_libs/blob/9de06305edb99bbe73be1094ed1b9bc47679b854/libsel4vm/src/arch/arm/vm.c#L256-L259

**Comment by axel-h** [comment]
Ah yes, it's set there eventually for all errors.
